src/explorer_command.cc

```diff
@@ -85,35 +84,83 @@ namespace {
   return out;
 }
 
+static int IsContextMenuEnabled() {
+  static int enabled = -1;
+  HKEY subhkey;
+  int err;
+#if defined(INSIDER)
+    const wchar_t kTitleRegkey[] = L"Software\\Classes\\VSCodeInsidersContextMenu";
+#else
+    const wchar_t kTitleRegkey[] = L"Software\\Classes\\VSCodeContextMenu";
+#endif
+
+  if (enabled != -1)
+    return enabled;
+
+  // Check if the context menu is enabled in the registry.
+  err = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                      kTitleRegkey,
+                      0,
+                      KEY_QUERY_VALUE | KEY_WOW64_64KEY,
+                      &subhkey);
+  if (err != ERROR_SUCCESS) {
+    err = RegOpenKeyExW(HKEY_CURRENT_USER,
+                        kTitleRegkey,
+                        0,
+                        KEY_QUERY_VALUE | KEY_WOW64_64KEY,
+                        &subhkey);
+  }
+
+  if (err != ERROR_SUCCESS) {
+    enabled = 0;
+  } else {
+    enabled = 1;
+    RegCloseKey(subhkey);
+  }
+
+  return enabled;
+}
+
 }
 
 class __declspec(uuid(DLL_UUID)) ExplorerCommandHandler final : public RuntimeClass<RuntimeClassFlags<ClassicCom | InhibitRoOriginateError>, IExplorerCommand> {
  public:
   // IExplorerCommand implementation:
   IFACEMETHODIMP GetTitle(IShellItemArray* items, PWSTR* name) {
-    const size_t kMaxStringLength = 1024;
-    wchar_t value_w[kMaxStringLength];
-    wchar_t expanded_value_w[kMaxStringLength];
-    DWORD value_size_w = sizeof(value_w);
+    static std::wstring cached_title;
+    static bool title_cached = false;
+    
+    if (!title_cached) {
+      const size_t kMaxStringLength = 1024;
+      wchar_t value_w[kMaxStringLength];
+      wchar_t expanded_value_w[kMaxStringLength];
+      DWORD value_size_w = sizeof(value_w);
 #if defined(INSIDER)
-    const wchar_t kTitleRegkey[] = L"Software\\Classes\\VSCodeInsidersContextMenu";
+      const wchar_t kTitleRegkey[] = L"Software\\Classes\\VSCodeInsidersContextMenu";

```

Comment:
are these formatting glitch?

Reply:
nope, it is intended to align with the conditional block.